Training/Training/intlist.c

```diff
@@ -0,0 +1,188 @@
+#include "intlist.h"
+#include <malloc.h>
+#include<stdbool.h>
+
+struct LinkedList* Create () {
+   struct LinkedList* list = NULL;
+   do {
+      list = (struct LinkedList*)malloc (sizeof (struct LinkedList));
+      if (list != NULL) {
+         list->head = NULL;
+         list->isDeleted = false;
+         list->count = 0;
+      }
+   } while (list == NULL);
+   return list;
+}
+
+struct Node* CreateNode (int data) {
+   struct Node* newNode = NULL;
+   do {
+      newNode = (struct Node*)malloc (sizeof (struct Node));
+      if (newNode != NULL) {
+         newNode->data = data;
+         newNode->next = NULL;
+      }
+   } while (newNode == NULL);
+   return newNode;
+}
+
+int Delete (struct LinkedList* list) {
+   if (list->isDeleted == true)
+      return E_LIST_DNE;
+   if (list->head == NULL) {
+      list->isDeleted = true;
+      return 0;
+   }
+   struct Node* temp = NULL;

```

Comment:
The declaration of temp can be inside the "while" loop.
struct Node* temp = list->head;

Or even while((struct Node* temp = list->head) != null) {
..
..
}